.github/workflows/ci.yml

```diff
@@ -44,6 +42,38 @@ jobs:
         env:
           TEST_FASTTEXT: true
       - uses: codecov/codecov-action@v3
+  check-without-extras:
+    name: Check cleanlab runs without optional dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          python -m  pip install --upgrade pip
+          pip install .
+          pip install pytest pytest-lazy-fixture
+      - name: Run tests
+        run: | # these tests provide 50% coverage
+          pytest tests/test_multilabel_classification.py tests/test_multiannotator.py tests/test_filter_count.py
+  check-min-version:
+    name: Checks the package is compatible with minimum versions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.7'
+      - name: Install min version of dependencies
+        run: |
+          python -m  pip install --upgrade pip
+          pip install .
+          pip install pytest pytest-lazy-fixture numpy==1.20.0 scikit-learn==1.0 tqdm==4.53.0 pandas==1.1.5 termcolor==2.0.0
```

Comment:
are you sure this second install overwrites the dependencies versions if they were already installed during `pip install .` ?

Reply:
this is not super forwards compatible in the sense somebody will need to remember to update this every time the min versions are updated. I guess it's ok because presumably this part of CI would break (if the old min version no longer works) and then the developer will know they need to update these lines too.  But this check wouldn't be thorough anymore if somebody adds a new dependency with a minimum version and forgets to update CI, so at the very least need to ensure the pip install of cleanlab did not install any other dependencies. 

But stronger version of this would be to programmatically fetch what are the dependencies, what are their minimum versions, uninstall all dependencies, install min versions of them. Is that a lot of work though?

Reply:
Let me check if there's a command to get answers to these 2 questions: `what are the dependencies, what are their minimum versions,` rest of the part is easy

Reply:
Added code that automatically gets min versions of dependencies and installs them